src/librustc/diagnostics.rs

```diff
@@ -31,13 +47,51 @@ it is impossible to create an instance of an empty type, so empty match
 expressions are almost never desired.  This error is typically fixed by adding
 one or more cases to the match expression.
 
-An example of an empty type is `enum Empty { }`.
+An example of an empty type is `enum Empty { }`. So, the following will work:
+
+```
+fn foo(x: Empty) {
+    match x {
+        // empty
+    }
+}
+
+```
+
+but this won't:
```

Comment:
yeah it's... strange. English grammar people aren't generally paying attention to putting code blocks into sentences ;)